File: api/omp/codegen/lib/ompGetMaxThreads/ompGetMaxThreads.c
#include "ompGetMaxThreads.h"
#include "omp.h"

int ompGetMaxThreads(void)
{
  return omp_get_max_threads();
}

void ompGetMaxThreads_initialize(void)
{
}

void ompGetMaxThreads_terminate(void)
{
}

File: api/omp4m/ompGetMaxThreads.m
function n = ompGetMaxThreads 
%ompGetMaxThreads Returns the maximum value that can be returned by a call to the ompGetNumThreads function.
%   n = ompGetMaxThreads
% See also ompGetThreadLimit

%#codegen -args {}

n = int32(1);
if ~strcmp(coder.target,'rtw'); return; end

coder.cinclude('omp.h');
coder.inline('always');
coder.allowpcode('plain')

n = coder.ceval('omp_get_max_threads');
